### PR TITLE
Share action tests

### PR DESCRIPTION
**Pull request by mendelB**
Adding some tests for the good ole [ShareAction](https://github.com/DoSomething/phoenix-next/blob/dev/resources/assets/components/ShareAction/ShareAction.js) component!

Also added a (somewhat meager) Facebook global object mock, to help with testing that we're triggering an FB share modal, and tracking the (successful or canceled) share.

### CODE REVIEW

File: resources/assets/components/ShareAction/ShareAction.test.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import setFBshare from '../../__mocks__/facebookShareMock';
+
+describe('ShareAction component', () => {
+  const url = 'https://dosomething.org';
+  const trackingData = { url };
+
+  let trackEventMock = jest.fn();
+  let openModalMock = jest.fn();
+
+  const getShallow = () => shallow(
```
Comment (weerd) [inline resources/assets/components/ShareAction/ShareAction.test.js]: Just curious but would `beforeEach()` here make sense for creating this component before each test?

Comment (weerd) [reply resources/assets/components/ShareAction/ShareAction.test.js]: oh, nevermind, you use it later in the `describe()`

Comment (mendelB) [reply resources/assets/components/ShareAction/ShareAction.test.js]: Also what I found tricky about this is that we need a reference to the `shallow` object for testing, and I wasn't sure how to maintain a reference to it through `beforeEach`.

Comment (mendelB) [reply resources/assets/components/ShareAction/ShareAction.test.js]: Hmm... maybe by declaring it above it scope, and using a `let`? 🤔

Comment (mendelB) [reply resources/assets/components/ShareAction/ShareAction.test.js]: That worked! And I think it's slightly more elegant. I'll update this and you can confirm

Comment (weerd) [reply resources/assets/components/ShareAction/ShareAction.test.js]: Noice! I also wonder if [`beforeAll()`](https://facebook.github.io/jest/docs/en/setup-teardown.html#one-time-setup) would help with the scope issue 🤔

Comment (mendelB) [reply resources/assets/components/ShareAction/ShareAction.test.js]: Oh wait yes!

Comment (mendelB) [reply resources/assets/components/ShareAction/ShareAction.test.js]: OK @weerd sorry I got a little antsy here. But I *think* what we have based on your feedback is an improvement.

**Pull request merged**